samples/simple_vector_add/simple_vector_add.cpp

```diff
@@ -70,9 +72,13 @@ int main() {
   simple_vadd<float, array_size>(D, E, F);
   for (unsigned int i = 0; i < array_size; i++) {
     if (C[i] != A[i] + B[i]) {
+      std::cout << "The results are incorrect (element " << i << " is "
+                << F[i] << "!\n";
       return 1;
```

Comment:
This will exit after the first error, which is fine, you could add a bool variable that represents success or failure and return that at the end of execution. Something like
```c++
bool failure = false;
for (i < arr.size()) {
  if(C[i] != A[i] + B[i]) {
    failure = true;
    std::cout << "fail message";
  }
}
return failure;
```